### PR TITLE
Wpscan Bug Fix 

### PR DESCRIPTION
**Pull request by zkkok**
# Changes 

- **Reduced Code Redundancy:** Replaced some unnecessary checkboxes with NativeSelect and TextInput
- **Custom Mode:** Removed some uncommon features and added a custom configuration option to make the GUI look less packed
- **Tooltip and Placeholder Descriptions:** Corrected the grammatical error in the tooltip and added some placeholder descriptions to  better describe each option

Link issue to #415

### CODE REVIEW

**Comment by bradleybogg** [comment]
@zkkok - I found this useful https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/
I have learnt some information here too that would be beneficial in future commits. I will look to implement in my work also.

**Comment by gangamarx** [comment]
During my initial review, I came across some error messages in the output, although the tooltips seemed fine. Therefore, I'm updating the DDT to make sure it's free of errors.

**Comment by zkkok** [comment]
@gangamarx could you please provide more details about the error u encountered?

**Comment by gangamarx** [comment]
WPScan test version: 5deef3c - Bradley Bogg, 2 days ago : Merge pull request #524
1. Tooltip reads well and error free. I made a suggestion though!. See attached file
2. I scanned the URL https://sites.lsa.umich.edu/julian/ and the test results are great-successful. See attached a file
3. In the advanced mode there is spelling error "Login URL". See attached file
[1-Suggestion for Tooltip.docx](https://github.com/Hardhat-Enterprises/Deakin-Detonator-Toolkit/files/14930352/1-Suggestion.for.Tooltip.docx)
2-WPScan output review
3-Advanced Mode spelling mistake-login URL

**Comment by gangamarx** [comment]
I didn't take a screen shot as I found the head was different, now I fixed that. Results are great. Should I be checking the advanced mode too ?

________________________________

**Comment by bradleybogg** [comment]
@gangamarx, test the entire tool and compare against what is requested in the card. All PRs are linked to an issue which describes what should have been changed.

**Comment by gangamarx** [comment]
Apologies for the delay; I was thoroughly reviewing the results from the advanced options. I've included a summary in document format and added screenshots for clarity. While most commands functioned correctly in CLI mode, only 3 out of 41 worked in GUI mode. I conducted the tests to the best of my abilities with the knowledge I have and am submitting this review accordingly.
[WPscan Advanced mode -test results.docx](https://github.com/Hardhat-Enterprises/Deakin-Detonator-Toolkit/files/14975327/WPscan.Advanced.mode.-test.results.docx)

**Review by gangamarx — COMMENTED** [review]
Apologies for the delay; I was thoroughly reviewing the results from the advanced options. I've included a summary in document format and added screenshots for clarity. While most commands functioned correctly in CLI mode, only 3 out of 41 worked in GUI mode. I conducted the tests to the best of my abilities with the knowledge I have and am submitting this review accordingly.
[WPscan Advanced mode -test results.docx](https://github.com/Hardhat-Enterprises/Deakin-Detonator-Toolkit/files/14975338/WPscan.Advanced.mode.-test.results.docx)

**Comment by zkkok** [comment]
Hi @gangamarx,

Thanks for carrying out the functionality tests. However, based on the document you provided it seems that you are performing the tests on the old version of the tool, please switch to the WPSScan-Fix branch and perform the tests based on the updated code.

**Comment by gangamarx** [comment]
Thanks @zkkok , I apologize wasting time working on the incorrect branch. I will work on the right one now and send the results asap. Cheers

**Comment by bradleybogg** [comment]
@gangamarx please conduct application testing.

**Comment by gangamarx** [comment]
Conducted the application testing on new WPSscan-Fix branch. All tests were successful. Attached is the evidence of test.
[WPSScan-Fix-Advanced testing review.docx](https://github.com/Hardhat-Enterprises/Deakin-Detonator-Toolkit/files/15019286/WPSScan-Fix-Advanced.testing.review.docx)

**Review by gangamarx — COMMENTED** [review]
Conducted the application testing on new WPSscan-Fix branch. All tests were successful. Attached is the evidence of test.
[WPSScan-Fix-Advanced testing review.docx](https://github.com/Hardhat-Enterprises/Deakin-Detonator-Toolkit/files/15019286/WPSScan-Fix-Advanced.testing.review.docx)

**Comment by gangamarx** [comment]
No grammatical errors in the tool tip. All looks okay.

File: src/components/WPScan/WPScan.tsx [hunk 1]
```diff
@@ -16,153 +26,66 @@ const description_userguide =
...
+    "All themes",
+    "Popular themes",
+    "Timthumbs",
+    "Config Backups",
+    "Db exports",
+    "UID range",
+    "MID range",
+    "Custom",
+];
+const enumerationtypesrequiringupdownbound = ["UID range", "MID range"];
```
Comment (bradleybogg) [inline src/components/WPScan/WPScan.tsx:45]: - [ ] Camelcase required
- [ ] Simplify variable naming

File: src/components/WPScan/WPScan.tsx [hunk 2]
```diff
@@ -16,153 +26,66 @@ const description_userguide =
...
+const detectionmodes = ["mixed", "passive", "aggressive"];
```
Comment (bradleybogg) [inline src/components/WPScan/WPScan.tsx:46]: - [ ] camelCase

File: src/components/WPScan/WPScan.tsx [hunk 3]
```diff
@@ -16,153 +26,66 @@ const description_userguide =
...
+const outputformats = ["cli-no-colour", "cli-no-color", "json", "cli"];
```
Comment (bradleybogg) [inline src/components/WPScan/WPScan.tsx:47]: - [ ]  camelCase

File: src/components/WPScan/WPScan.tsx [hunk 4]
```diff
@@ -16,153 +26,66 @@ const description_userguide =
...
 
 interface FormValues {
     url: string;
-    optionBlank: string;
+    lowbound: number;
```
Comment (bradleybogg) [inline src/components/WPScan/WPScan.tsx:51]: - [ ]  camelCase

File: src/components/WPScan/WPScan.tsx [hunk 5]
```diff
@@ -16,153 +26,66 @@ const description_userguide =
...
+    upbound: number;
```
Comment (bradleybogg) [inline src/components/WPScan/WPScan.tsx:52]: - [ ]  camelCase

File: src/components/WPScan/WPScan.tsx [hunk 6]
```diff
@@ -16,153 +26,66 @@ const description_userguide =
...
+    customenum: string;
```
Comment (bradleybogg) [inline src/components/WPScan/WPScan.tsx:53]: - [ ]  camelCase

File: src/components/WPScan/WPScan.tsx [hunk 7]
```diff
@@ -16,153 +26,66 @@ const description_userguide =
...
     stealthy: boolean;
+    custom: string;
 }
 
 const WPScan = () => {
     const [loading, setLoading] = useState(false);
     const [output, setOutput] = useState("");
     const [checkedAdvanced, setCheckedAdvanced] = useState(false);
+    const [checkedCustom, setcheckedCustom] = useState(false);
+    const [selectedenumerationtype, setselectedenumerationtype] = useState("");
```
Comment (bradleybogg) [inline src/components/WPScan/WPScan.tsx:68]: - [ ]  camelCase

File: src/components/WPScan/WPScan.tsx [hunk 8]
```diff
@@ -16,153 +26,66 @@ const description_userguide =
...
-    loginUri: string;
     stealthy: boolean;
+    custom: string;
 }
 
 const WPScan = () => {
     const [loading, setLoading] = useState(false);
     const [output, setOutput] = useState("");
     const [checkedAdvanced, setCheckedAdvanced] = useState(false);
+    const [checkedCustom, setcheckedCustom] = useState(false);
```
Comment (bradleybogg) [inline src/components/WPScan/WPScan.tsx:67]: - [ ]  camelCase

File: src/components/WPScan/WPScan.tsx [hunk 9]
```diff
@@ -16,153 +26,66 @@ const description_userguide =
...
+    const [selectedenumerationtype, setselectedenumerationtype] = useState("");
+    const [selecteddetectionmode, setselecteddetectionmode] = useState("");
```
Comment (bradleybogg) [inline src/components/WPScan/WPScan.tsx:69]: - [ ]  camelCase

File: src/components/WPScan/WPScan.tsx [hunk 10]
```diff
@@ -16,153 +26,66 @@ const description_userguide =
...
+    const [selectedoutputformat, setselectedoutputformat] = useState("");
```
Comment (bradleybogg) [inline src/components/WPScan/WPScan.tsx:70]: - [ ]  camelCase

**Review by bradleybogg — CHANGES_REQUESTED** [review]
Wow good work simplifying this component was previously very poorly written.
Can you please update the variable naming please.

**Review by bradleybogg — APPROVED** [review]

**Pull request merged**